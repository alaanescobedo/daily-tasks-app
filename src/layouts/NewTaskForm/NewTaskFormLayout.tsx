import { AppContainerLayout } from '@layouts'
import styles from './NewTaskFormLayout.module.css'

export interface NewTaskFormLayoutProps {
  children: React.ReactNode
  id: string
  handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void
  title?: string
  sendLabel?: string
}

export const NewTaskFormLayout = ({ children, id, handleSubmit, title = 'New Task', sendLabel = 'Send' }: NewTaskFormLayoutProps): JSX.Element => {
  return (
    <AppContainerLayout>
      <h1 className={styles.heading}>{title}</h1>
      <form className={styles.form} onSubmit={handleSubmit} id={id}>
        {children}
        <button type='submit' className={styles.input_submit}>{sendLabel}</button>
      </form>
    </AppContainerLayout>
  )
}
