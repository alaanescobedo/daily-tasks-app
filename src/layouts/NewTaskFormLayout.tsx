import { AppContainerLayout } from './AppContainerLayout'
import styles from './NewTaskFormLayout.module.css'

interface Props {
  children: React.ReactNode
  handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void
  title?: string
  sendLabel?: string
}

export const NewTaskFormLayout = ({ children, handleSubmit, title = 'New Task', sendLabel = 'Send' }: Props): JSX.Element => {
  return (
    <AppContainerLayout>
      <h1 className={styles.heading}>{title}</h1>
      <form className={styles.form} onSubmit={handleSubmit}>
        {children}
        <button type='submit' className={styles.input_submit}>{sendLabel}</button>
      </form>
    </AppContainerLayout>
  )
}
