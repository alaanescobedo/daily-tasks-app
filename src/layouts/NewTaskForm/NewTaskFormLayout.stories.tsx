import { ComponentStory, ComponentMeta } from '@storybook/react'

import { NewTaskFormLayout as NewTaskFormLayoutComponent, NewTaskFormLayoutProps } from './NewTaskFormLayout'

import styles from '../../App.module.css'

const NewTaskFormLayoutStory: ComponentMeta<typeof NewTaskFormLayoutComponent> = {
  title: 'Layout/New Task Form',
  component: NewTaskFormLayoutComponent
}
export default NewTaskFormLayoutStory

const Template: ComponentStory<typeof NewTaskFormLayoutComponent> = (args) => (
  <span className={styles.app_container}>
    <NewTaskFormLayoutComponent {...args} />
  </span>
)

export const NewTaskForm = Template.bind({})
const NewTaskFormLayoutArgs: NewTaskFormLayoutProps = {
  children: (
    <span style={{ color: 'white', gridColumn: '1 / -1', textAlign: 'center' }}>
      ʕ•́ᴥ•̀ʔっayo!
    </span>
  ),
  handleSubmit: () => { },
  title: 'New Task'
}
NewTaskForm.args = NewTaskFormLayoutArgs
