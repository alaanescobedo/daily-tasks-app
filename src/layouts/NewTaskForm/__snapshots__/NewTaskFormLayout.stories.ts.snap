// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/New Task Form New Task Form 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <span
      className="app_container"
    >
      <div
        className="container"
      >
        <h4
          className="username"
        >
          Guest
        </h4>
        <h4
          className="date"
        >
          Tuesday,  4/12/2022
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            Clock
          </div>
        </h4>
      </div>
      <div
        className="super_container"
      >
        <div
          className="container"
        >
          <h1
            className="heading"
          >
            New Task
          </h1>
          <form
            className="form"
            onSubmit={[Function]}
          >
            <span
              style={
                Object {
                  "color": "white",
                  "gridColumn": "1 / -1",
                  "textAlign": "center",
                }
              }
            >
              ʕ•́ᴥ•̀ʔっayo!
            </span>
            <button
              className="input_submit"
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
        <div
          className="return_container"
        >
          <a
            aria-current="page"
            className="return active"
            href="/"
            onClick={[Function]}
          >
            Return
          </a>
        </div>
      </div>
    </span>
     
  </div>,
  " ",
]
`;
